src/main/java/org/wso2/securevault/secret/SecretManager.java

```diff
@@ -214,6 +191,51 @@ public void init(Properties properties) {
         initialized = true;
     }
 
+    /**
+     * Check whether to use the provider listed under secretRepositories
+     * or secretProviders property for resolving secrets
+     *
+     * @param secretAnnotation String contains the alias, the provider type and the repository type
+     * @return plain text value for the required secret
+     */
+    public String resolveSecret(String secretAnnotation) {
+
+        String[] secretAnnotationStrings = secretAnnotation.split(DELIMITER);
+        if (secretAnnotationStrings.length == 1) {
+            if (isLegacyProvidersExists) {
+                return getSecret(secretAnnotation);
+            }
+        }
+        return resolveSecret(secretAnnotationStrings);
+    }
+
+    /**
+     * Resolve the secret annotation for the secrets coming from
+     * repositories belongs to providers listed under secretProviders property
+     *
+     * @param annotation vaule retrieve by the resolveSecret as the value to be resolved
+     * @return If there is a secret , otherwise , alias itself
+     */
+    public String resolveSecret(String[] annotation) {
+
+        int length = annotation.length;
+        try {
+            switch (length) {
+                case 1:
+                    return getSecret((String) providers.keySet().toArray()[0],
+                            (String) secretRepositories.keySet().toArray()[0],
+                            annotation[0]);
+                case 3:
+                    return getSecret(annotation[0], annotation[1], annotation[2]);
+                default:
+                    throw new IllegalArgumentException("invalid annotation");
+            }
+        } catch (NullPointerException e) {
+            handleException("No Secret Repositories have been initialized : ");
```

Comment:
```suggestion
            handleException("No Secret Repositories have been initialized: ");
```

Reply:
e is not passed to handleException method

Reply:
Remove NPE catch and replaced it with a null check